[Darien Tan Wei Hao] iP

reven0n opened the pull request.

vincent-wat04 commented on src/main/java/som/command/FindCommand.java:
```diff
@@ -0,0 +1,39 @@
+     *
+     * @param dateStr the date to search for; must not be null.
+     */
+    public FindCommand(String dateStr) {
```
FindCommand should refer to finding certain tasks whose descriptions include the keyword, instead of finding tasks on a specific data, which I think is already featured in another function.

vincent-wat04 commented on src/main/java/som/task/Deadline.java:
```diff
@@ -0,0 +1,64 @@
+     *
+     * @return the LocalDateTime representing when the task is due.
+     */
+    public LocalDateTime getby() {
```
Maybe it's better to name this function 'getBy()'?

vincent-wat04 commented on src/main/java/som/command/Command.java:
```diff
@@ -0,0 +1,37 @@
+import som.TaskList;
+import som.Ui;
+
+/**
+ * Defines the interface for executable commands in the Som chatbot.
+ * <p>This abstract class ensures that all command objects can be executed
+ * uniformly by the main application loop. Each concrete command must
+ * implement the execute method to perform its specific action and indicate
+ * whether it triggers application exit.</p>
+ *
+ * @author Darien Tan
+ */
```
The JavaDoc is well written and covers all classes and methods 👍👍👍

vincent-wat04 commented on src/main/java/som/Parser.java:
```diff
@@ -0,0 +1,148 @@
+        String command = parts[0].toLowerCase();
+        String fullDesc = parts.length > 1 ? parts[1] : "";
+
+        switch (command) {
+        case "bye": {
+            return new ExitCommand();
+        }
+        case "list": {
+            return new ListCommand();
+        }
```
Very nice! follows the indentation rule for switch statements👏👍

vincent-wat04 reviewed (commented):
The code consistently follows the coding standard and the implementation is well structured and clean. Good job!👍

sngdarren commented on src/main/java/som/command/AddCommand.java:
```diff
@@ -0,0 +1,44 @@
+    }
+
+    @Override
+    public void execute(TaskList tasks, Ui ui) throws SomException {
```
maybe tasks here should be taskList instead for readability

sngdarren commented on src/main/java/som/task/Deadline.java:
```diff
@@ -0,0 +1,63 @@
+     *
+     * @return the LocalDateTime representing when the task is due.
+     */
+    public LocalDateTime getby() {
```
Perhaps it would be better to name this getBy() to adhere to camel case

sngdarren commented on src/main/java/som/command/UnmarkCommand.java:
```diff
@@ -0,0 +1,42 @@
+    private int index;
+
+    /**
+     * Constructs a MarkCommand with the given task index.
```
Could this me a mistake - should it be UnmarkCommand instead

reven0n commented:
Addressed comments in PR review and made relevant changes at [8968a40](https://github.com/nus-cs2103-AY2526S1/ip/pull/326/commits/8968a4080f2b0bf6fd75310255b9d10529b32958)
